src/debug/telemetry/telemetry2_slot.h

```diff
@@ -0,0 +1,72 @@
+/* SPDX-License-Identifier: Apache-2.0
+ * Copyright (c) 2024 Intel Corporation
+ */
+
+#ifndef _ZEPHYR_SOC_INTEL_ADSP_TELEMETRY2_SLOT
+#define _ZEPHYR_SOC_INTEL_ADSP_TELEMETRY2_SLOT
+
+/*
+ * Telemetry2 slot is taken from SRAM window divided into several
+ * chunks with simple header declared bellow.
+ *
+ * "type" describes the contents of the chunk
+ * "size" describes the size of the chunk in bytes
+ *
+ * After the chunk another chunk is expected to follow, if there is
+ * enough space for another header. If type is 0 and size 0 it
+ * indicated that the rest of the slot is unused, and this header is
+ * free for taking.
+ *
+ * Each chunk is aligned to start at 64-byte cache line boundary.
+ *
+ * For example:
+ *      --------------------------------------------------  ---
+ *      | magic = TELEMETRY2_PAYLOAD_MAGIC               |   |
+ *      | hdr_size = 64                                  |   |
+ *      | total_size = 320                               |   |
+ *      | abi = TELEMETRY2_PAYLOAD_V0_0                  |  64 bytes
+ *      | tstamp = <aligned with host epoch>             |   |
+ *      | <padding>                                      |   |
+ *	--------------------------------------------------  ---
+ *	| type = ADSP_DW_TELEMETRY2_ID_THREAD_INFO       |   |
+ *      | size = 256                                     |  256 bytes
```

Comment:
can the size be different from a multiple of 64? IOW, can we have gaps due to alignment?

Reply:
Yes, that is my intention. Thou, when I think about it I am not sure if this is a necessary requirement. I just had so many problems in the beginning with cache pecularities, I played it safe. But then again the debug window pointers are all uncached pointers, so maybe the system would survive two cores writing concurrently in the same cache line. I'll check.